Libraries/LibGUI/LinkLabel.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, Hüseyin ASLITÜRK <asliturk@hotmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <LibGUI/Label.h>
+
+namespace GUI {
+
+class LinkLabel : public Label {
+    C_OBJECT(LinkLabel)
+public:
+    virtual ~LinkLabel() override;
+
+    void set_href(const StringView& value) { m_href = value; }
+
+protected:
+    explicit LinkLabel(const StringView& text = {});
+
+    virtual void enter_event(Core::Event&) override;
+    virtual void leave_event(Core::Event&) override;
+    virtual void mousedown_event(MouseEvent&) override;
+    virtual void paint_event(PaintEvent&) override;
+
+private:
+    String m_href;
```

Comment:
What's the difference between this and the "text"? Aren't they always set to the same value?

Reply:
Hi,
In general use, the text is the same as href, so the constructor sets the text to href.
If you want to set a different href than the text, there is the `set_href `method.
```
auto& link_label = right_container.add<LinkLabel>("Web Site");
link_label.set_href("http://serenityos.org");
```

Reply:
Ah, that makes sense. Perhaps let's add a comment about it?